proteus/Domain.py

```diff
@@ -301,6 +302,45 @@ def writeAsymptote(self, fileprefix):
     def writeXdmf(self,ar):
         raise UserWarning("Xdmf output not implemented")
 
+def unitSimplex(nd=2):
+    """Builds a 2 or 3 dimension reference element.
+    
+    Parameters
+    ----------
+    nd : int
+        The elements dimension (must be 2 or 3)
+
+    Returns
+    -------
+    reference_element
+    """
+    if nd!=2 and nd!=3:
+        logEvent("ERROR - Reference element must have dimension 2 or 3")
+        sys.exit(1)
```

Comment:
It's nice to have logEvents, because that allows them to be very visible in the log file.  It's very handy to have ValueErrors or other errors built off of the various exception classes, though, because we can catch and handle them in other parts of code.  

For example, we could have a statement which tries to unitSimplex(nd) if possible, but has a backup plan if it isn't able to - it can catch a specific ValueError and use that to change its behavior without necessarily crashing (while crashing if other things break).  This also makes it easy to detect if functions are throwing the correct errors in tests, since we can send in wrong results and make sure it breaks the way we want it to.  Note that you can (and should) add in messages to errors, and can even develop simple error subclasses if there's a particular type of error you want to keep separate from the rest.

It might be useful to have a logEvent/ValueError (or, again, any other type of error) combination, if we want to make sure the error gets written down.  Alternatively, having a single function which raises errors and also writes a log event might be nice (I guess you'd pass the specific error type as a function, because there isn't a fixed set of potential errors?), or having some way of catching errors and printing them in the log as if they were logEvents... I don't know how the logging code works, nor if it or python's exceptions have any good qualities for parallelism (the main concern I'd have is that one or the other might be more prone to a race condition which garbles its output if several nodes fail).